core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,247 @@
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))
+        }
+      }
+  }
+  
+  object SetACLCallback extends StatCallback {
+    def processResult(rc: Int,
+                      path: String,
+                      ctx: Object,
+                      stat: Stat) {
+      val zkHandle = zkUtils.zkConnection.getZookeeper
+      val promise = ctx.asInstanceOf[Promise[String]]
+      
+      Code.get(rc) match {
+          case Code.OK =>
+            info("Successfully set ACLs for %s".format(path))
+            promise success "done"
+          case Code.CONNECTIONLOSS =>
+            Future {
+              zkHandle.setACL(path, ZkUtils.DefaultAcls(zkUtils.isSecure), -1, SetACLCallback, ctx)
+            }(threadExecutionContext)
+          case Code.NONODE =>
+            warn("Node is gone, it could be have been legitimately deleted: %s".format(path))
+          case Code.SESSIONEXPIRED =>
+            // Starting a new session isn't really a problem, but it'd complicate
+            // the logic of the tool, so we quit and let the user re-run it.
+            error("ZooKeeper session expired while changing ACLs")
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            promise failure ZkException.create(KeeperException.create(Code.get(rc)))   
+      }
+    }
+  }
+
+  def run(args: Array[String]) {
+    var jaasFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
+    val parser = new OptionParser()
+
+    val goOpt = parser.accepts("kafka.go", "Indicates whether to make this cluster secure or unsecure. The "
+        + " options are 'secure' and 'unsecure'").withRequiredArg().ofType(classOf[String])
+    val jaasFileOpt = parser.accepts("jaas.file", "JAAS Config file.").withOptionalArg().ofType(classOf[String])
+    val zkUrlOpt = parser.accepts("zookeeper.connect", "Sets the ZooKeeper connect string (ensemble). This parameter " +
+      "takes a comma-separated list of host:port pairs.").withRequiredArg().defaultsTo("localhost:2181").
+      ofType(classOf[String])
+    val zkSessionTimeoutOpt = parser.accepts("zookeeper.session.timeout", "Sets the ZooKeeper session timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val zkConnectionTimeoutOpt = parser.accepts("zookeeper.connection.timeout", "Sets the ZooKeeper connection timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val helpOpt = parser.accepts("help", "Print usage information.")
+
+    val options = parser.parse(args : _*)
+    if (options.has(helpOpt))
+      CommandLineUtils.printUsageAndDie(parser, usageMessage)
+
+    if ((jaasFile == null) && !options.has(jaasFileOpt)) {
+      error("No JAAS configuration file has been specified. Please make sure that you have set either "
```

Comment:
For all those error and info logging, if they are meant for the users to see, we should print them out to stdout instead of logging since the log4j may be turned off when running the tools.